CANserver/Logging.cpp

```diff
@@ -290,6 +320,54 @@ void CANServer::Logging::handleMessage(CAN_FRAME *frame, const uint8_t busId)
                     break;
                 }
 
+                case CANServer::Logging::LogType_Drive:
+                {
+                    long currentLogMillis = millis();
+                    int gearNUM = 0;
+                    String tempRow = String(_currentTimeOfDay.tv_sec);
+                    CANServer::CanBus *canbusInstance = CANServer::CanBus::instance();
+                    CANServer::CanBus::AnalysisItemMap::const_iterator found_it = canbusInstance->dynamicAnalysisItems()->find("Gear");
+                    gearNUM = found_it->second->lastValue;
+
+                    if((currentLogMillis - previousDriveLogcycle >= DRIVELOGGING_INTERVAL) && (gearNUM == 2 || gearNUM == 4))
+                    {
+                        previousDriveLogcycle = currentLogMillis;
+                        CANServer::CanBus *canbusInstance = CANServer::CanBus::instance();
```

Comment:
You are grabbing the instance of canbus class twice here.  Extra code that doesn't need to exist.